src/plugins/calculate/calculate.c

```diff
@@ -0,0 +1,331 @@
+
+	PNElem *stackPtr = stack;
+	PNElem result;
+	Operation resultOp = ERROR;
+	result.op = ERROR;
+	int ret;
+	if((ret = regcomp(&regex, regexString, REG_EXTENDED|REG_NEWLINE)))
+	{
+		ksDel(ks);
+		return result;
+	}
+	regmatch_t match;
+	int nomatch;
+	int start;
+	int len;
+	char *searchKey = NULL;
+	while(1)
+	{
+		nomatch = regexec(&regex, ptr, 1, &match, 0);
+		if(nomatch)
+		{
+			break;
+		}
+		len = match.rm_eo - match.rm_so;
+		start = match.rm_so + (ptr - prefixString);
+		if(len == 1)
+		{
+			switch(prefixString[start])
+			{
+				case '+':
+					stackPtr->op = ADD;
+					++stackPtr;
+					break;
+				case '-':
+					stackPtr->op = SUB;
+					++stackPtr;
+					break;
+				case '/':
+					stackPtr->op = DIV;
+					++stackPtr;
+					break;
+				case '*':
+					stackPtr->op = MUL;
+					++stackPtr;
+					break;
+				case '=':
+					if(resultOp == LT)
+						resultOp = LE;
+					else if(resultOp == GT)
+						resultOp = GE;
+					else if(resultOp == ERROR)
+						resultOp = EQU;
+					else if(resultOp == EQU)
+						resultOp = EQU;
+					else if(resultOp == NOT)
+						resultOp = NOT;
+					break;
+				case '<':
+					resultOp = LT;
+					break;
+				case '>':
+					resultOp = GT;
+					break;
+				case '!':
+					resultOp = NOT;
+					break;
+				default:
+					ELEKTRA_SET_ERRORF(122, parentKey, "%c isn't a valid operation", prefixString[start]);
+					regfree(&regex);
+					if(searchKey)
+						free(searchKey);
+					free(stack);
+					ksDel(ks);
+					return result;	
+					break;
+			}
+		}
+		else
+		{
+			ksRewind(ks);
+			searchKey = realloc(searchKey, len+2+strlen(keyName(parentKey)));
+			strcpy(searchKey, keyName(parentKey));
+			strcat(searchKey, "/");
+			strncat(searchKey, ptr+1, len);
+			key = ksLookupByName(ks, searchKey, 0);
+			if(!key)
+			{
+				ELEKTRA_SET_ERRORF(124, parentKey, "Couldn't find Key %s", searchKey);
+				regfree(&regex);
+				free(searchKey);
+				ksDel(ks);
+				free(stack);
+				return result;
+			}
+			stackPtr->value = atof(keyString(key));
+			stackPtr->op = VAL;
+			++stackPtr;
+		}
+		int offset = stackPtr - stack;
+		stack = realloc(stack, (offset + 1)*sizeof(PNElem));
+		stackPtr = stack;
+		stackPtr += offset;
+		ptr += match.rm_eo;
+	}
+	regfree(&regex);
+	free(searchKey);
+	ksDel(ks);
+	stackPtr->op = END;
+	result = doPrefixCalculation(stack, stackPtr);	
+	if(result.op != ERROR)
+		result.op = resultOp;
+	free(stack);
+	return result;
+}
+
+int elektraCalculateSet(Plugin *handle ELEKTRA_UNUSED, KeySet *returned ELEKTRA_UNUSED, Key *parentKey ELEKTRA_UNUSED)
+{
+	/* set all keys */
+	Key *cur;
+	const Key *meta;
+	PNElem result;
+	while((cur = ksNext(returned)) != NULL)
+	{
+		meta = keyGetMeta(cur, "check/calc");
+		if(!meta)
+			continue;
+		result = parsePrefixString(keyString(meta), ksDup(returned), parentKey);
+		if(result.op == ERROR)
+		{
+			return 1;
+		}
+		else if(result.op == EQU)
+		{
+			if(fabs(atof(keyString(cur)) - result.value) > 0.00001)
```

Comment:
The epsilon is only there to prevent gcc from complaining about unsafe floating point comparison.
Gonna use a constant, but i don't think it should be configureable.